diff --git a/dist/src/lib/start/loadMetroConfig.js b/dist/src/lib/start/loadMetroConfig.js
index 0baf8126f0d4bc82928ece9d47cd205f2fc44ce6..8a714d50663df429f3ed9d39300248fd58ccf212 100644
--- a/dist/src/lib/start/loadMetroConfig.js
+++ b/dist/src/lib/start/loadMetroConfig.js
@@ -34,6 +34,7 @@ function getOverrideConfig(ctx, config) {
             // We can include multiple copies of InitializeCore here because metro will
             // only add ones that are already part of the bundle
             getModulesRunBeforeMainModule: () => [
+                ...(config.serializer?.getModulesRunBeforeMainModule?.() || []),
                 require.resolve(path.join(ctx.reactNativePath, 'Libraries/Core/InitializeCore'), { paths: [ctx.root] }),
                 ...outOfTreePlatforms.map((platform) => require.resolve(
                 // @ts-expect-error - TBD
