diff --git a/Libraries/Utilities/HMRClient.js b/Libraries/Utilities/HMRClient.js
index 776056cf527553fb659c0de31ab376342deddb10..6666c4ea2aae14c8bd73aeb73031b3a9de2a5afc 100644
--- a/Libraries/Utilities/HMRClient.js
+++ b/Libraries/Utilities/HMRClient.js
@@ -100,6 +100,9 @@ const HMRClient: HMRClientNativeInterface = {
   },
 
   registerBundle(requestUrl: string) {
+    const host = hmrClient._ws.url.split('/hot')[0];
+    // skip registerBundle calls from other hosts
+    if (!requestUrl.startsWith(host)) return;
     invariant(hmrClient, 'Expected HMRClient.setup() call at startup.');
     pendingEntryPoints.push(requestUrl);
     registerBundleEntryPoints(hmrClient);
@@ -167,6 +170,7 @@ const HMRClient: HMRClientNativeInterface = {
     hmrClient = client;
 
     const {fullBundleUrl} = getDevServer();
+
     pendingEntryPoints.push(
       // HMRServer understands regular bundle URLs, so prefer that in case
       // there are any important URL parameters we can't reconstruct from
